localstack/services/plugins.py

```diff
@@ -293,11 +293,39 @@ def get_service(self, name: str) -> Optional[Service]:
         return container.service if container else None
 
     def add_service(self, service: Service) -> bool:
-        state = ServiceState.AVAILABLE if service.is_enabled() else ServiceState.DISABLED
-        self._services[service.name()] = ServiceContainer(service, state)
+        self._mutex.acquire(blocking=True)
 
+        try:
+            state = ServiceState.AVAILABLE if service.is_enabled() else ServiceState.DISABLED
+            self._services[service.name()] = ServiceContainer(service, state)
+        except Exception as ex:
+            LOG.warning(f"Could not add service {service}", ex)
+
+        self._mutex.release()
```

Comment:
a good pattern for using the lock is leveraging the context manager:

```python
with self._mutex:
    ...
```
no manual releasing necessary